assignment7.txt

```diff
@@ -0,0 +1,18 @@
+Aly Milich
+
+1. ++*p and *++p both increment the value of the array that the pointer is pointing to whereas *p++ moves the pointer over.
+
+2. Left to right.
+
+3. Pointers make it easier to use arrays to change the values of certain spots. Since you don't need the for loop, it decreases the algorithmic complexity of your code and takes up less space in the computer & is easier to run.
+
+4.1 char array
+4.2 Invalid -- "xyz" is not in the array
+4.3 Invalid -- \0 will never equal 0
+4.4 Pointer
+4.5 Address is 0
```

Comment:
```This actually returns the type int*. Recall that a[0] would be of type int and the ampersand just returns the memory address. So &a[0] would return the pointer to the address (since a pointer is basically an address, which is why all pointers are 8 bytes no matter what they're pointing to), which would be of type int*. ```